tests/integration/_experimental/dataset/test_evaluation.py

```diff
@@ -69,12 +70,37 @@ def test__test() -> None:
 
     df_result = get_df_result()
     result_columns = ["softmax_bitmap", "score"]
-
     test(
         dataset_name,
         model_name,
         df_result,
-        on=JOIN_COLUMN,
+    )
+
+    fetched_df_dp, df_results_by_eval = fetch_results(dataset_name, model_name)
+    eval_cfg, fetched_df_result = df_results_by_eval[0]
+    assert len(df_results_by_eval) == 1
+    assert eval_cfg is None
+    expected_df_dp = df_dp[3:10].reset_index(drop=True)
+    expected_df_result = df_result.drop(columns=[JOIN_COLUMN])[3:10].reset_index(drop=True)
+    _assert_frame_equal(fetched_df_dp, expected_df_dp, dp_columns)
+    _assert_frame_equal(fetched_df_result, expected_df_result, result_columns)
+
+
+def test__test__iterator_input() -> None:
+    dataset_name = with_test_prefix(f"{__file__}::test__test")
+    model_name = with_test_prefix(f"{__file__}::test__test")
```

Comment:
names should be consistent with func name

Reply:
will do